spec/arrays.dd

```diff
@@ -192,6 +224,28 @@ assert(b[1] == 2);
 
     $(P See also $(GLINK2 expression, IndexExpression).)
 
+$(H3 $(LNAME2 pointer-arithmetic, Pointer Arithmetic))
+
+    $(P A pointer can also be indexed, but no bounds checks are done.
+    Unlike arrays, a pointer value can also be used in certain
+    arithmetic expressions to produce another pointer:)
+
+$(SPEC_RUNNABLE_EXAMPLE_RUN
+---
+int[] a = [1,2,3];
+int* p = a.ptr;
+
+*(p + 2) = 4; // same as `p[2] = 4`
+assert(a[2] == 4);
+
```

Comment:
Mention `ptrdiff_t` somewhere?

Reply:
I have added an *AddExpression* example showing that. I don't think we need it in the array docs, pointer indexing is enough IMO.